### PR TITLE
Drop the salttesting dependency.

### PR DESCRIPTION
**Pull request by s0undt3ch**
### What does this PR do?
* Drop the salttesting dependency cycle we currently have which we can't actually upgrade because of an unknown issue.
* Move the test daemons to the approach used by pytest-salt which is able to completely terminate all processes and respective children cleanly(this adds a dependency to pytest-salt, which will be a requirement when we completely move to pytest anyway).
* Absolute imports! No more `import integration`, `import unit`, just because we happened to inject the paths in `sys.path`, and, more significantly, no more `ensure_in_syspath`(which is most cases wasn't even being used properly).
* Code cleanup. On the move to pytest, we really don't need every module to have the `if __name__ == '__main__'` at the end of every test module.

### CODE REVIEW

**Comment by s0undt3ch** [comment]
Go Go Jenkins!

**Comment by cachedout** [comment]
@s0undt3ch If you can start on the lint errors, I will start looking at the rest. Thanks!

File: tests/support/parser/__init__.py [hunk 1]
```diff
@@ -0,0 +1,900 @@
...
+        '''
+
+    def finalize(self, exit_code=0):
+        '''
+        Run the finalization procedures. Show report, clean-up file-system, etc
+        '''
+        if self.options.no_report is False:
+            self.print_overall_testsuite_report()
+        self.post_execution_cleanup()
+        # Brute force approach to terminate this process and it's children
```
Comment (cachedout) [inline tests/support/parser/__init__.py]: `its children`

File: tests/integration/shell/test_master.py [hunk 2]
```diff
@@ -97,7 +95,7 @@ def test_exit_status_unknown_user(self):
             status, 'EX_NOUSER',
             message='unknown user not on system',
             stdout=stdout,
-            stderr=integration.utils.decode_byte_list(stderr)
+            stderr=tests.integration.utils.decode_byte_list(stderr)
```
Comment (s0undt3ch) [inline tests/integration/shell/test_master.py]: Removed that import.

**Comment by s0undt3ch** [comment]
Just force pushed the lint fixes

**Comment by cachedout** [comment]
@s0undt3ch Another merge conflict. :-/

**Comment by s0undt3ch** [comment]
Rebased. Let's check what jenkins has to say.

**Comment by s0undt3ch** [comment]
WooHoo! Everything passed!

**Pull request merged**